weatherreport.cpp

```diff
@@ -1,84 +1,59 @@
 #include <assert.h>
 #include <string>
 #include <iostream>
-
-using std::cout, std::endl, std::string;
-
+#include "./weatherreport.h"
+#include "./test-weather.h"
+using std::cout;
+using std::endl;
+using std::string;
 namespace WeatherSpace {
-class IWeatherSensor {
- public:
-        virtual double TemperatureInC() const = 0;
-        virtual int Precipitation() const = 0;
-        virtual int Humidity() const = 0;
-        virtual int WindSpeedKMPH() const = 0;
-};
-
-/// This is a stub for a weather sensor. For the sake of testing
-/// we create a stub that generates weather data and allows us to
-/// test the other parts of this application in isolation
-/// without needing the actual Sensor during development
-
-class SensorStub : public IWeatherSensor {
-    int Humidity() const override {
-        return 72;
+    SensorStub::SensorStub() {
```

Comment:
Good approach to make a data driven stub :)
But the stub is part of test code. Someone who takes your code to work with real sensors would be forced to take the test stub now